File: frontend/apps/pc-admin/smart-admin-web/src/api/permission.ts
import request, { type ApiResponse, type PageResponse } from '@/utils/request'

// 权限信息
export interface PermissionInfo {
  id: number
  tenantId: number
  parentId: number
  permissionName: string
  permissionCode: string
  permissionType: number
  path: string
  component: string
  icon: string
  apiPath: string
  method: string
  sort: number
  visible: number
  status: number
}

// 权限查询参数
export interface PermissionQuery {
  tenantId: number
  permissionName?: string
  permissionType?: number
  status?: number
  current?: number
  size?: number
}

// 权限API
export const permissionApi = {
  // 分页查询权限
  getPage(params: PermissionQuery): Promise<ApiResponse<PageResponse<PermissionInfo>>> {
    return request.get('/permission/page', { params })
  },

  // 获取权限树
  getTree(tenantId: number): Promise<ApiResponse<PermissionInfo[]>> {
    return request.get('/permission/tree', { params: { tenantId } })
  },

  // 获取权限详情
  getById(id: number): Promise<ApiResponse<PermissionInfo>> {
    return request.get(`/permission/${id}`)
  },

  // 创建权限
  create(data: Partial<PermissionInfo>): Promise<ApiResponse<number>> {
    return request.post('/permission', data)
  },

  // 更新权限
  update(id: number, data: Partial<PermissionInfo>): Promise<ApiResponse<void>> {
    return request.put(`/permission/${id}`, data)
  },

  // 删除权限
  delete(id: number): Promise<ApiResponse<void>> {
    return request.delete(`/permission/${id}`)
  },

  // 批量删除权限
  batchDelete(ids: number[]): Promise<ApiResponse<void>> {
    return request.delete('/permission/batch', { data: ids })
  },

  // 更新权限状态
  updateStatus(id: number, status: number): Promise<ApiResponse<void>> {
    return request.put(`/permission/${id}/status`, null, { params: { status } })
  },

  // 检查权限编码
  checkCode(code: string, tenantId: number, excludeId?: number): Promise<ApiResponse<boolean>> {
    return request.get('/permission/check-code', { params: { code, tenantId, excludeId } })
  }
}

export default permissionApi

File: backend/core/base/core-base/src/main/java/cn/aiedge/base/entity/Permission.java
package cn.aiedge.base.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 权限/菜单实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_permission")
public class Permission extends BaseEntity {

    /**
     * 权限编码（如：user:create、user:update）
     */
    private String permissionCode;

    /**
     * 权限名称
     */
    private String permissionName;

    /**
     * 权限类型（menu-菜单、button-按钮、api-API接口）
     */
    private String permissionType;

    /**
     * 父权限ID
     */
    private Long parentId;

    /**
     * 菜单路径
     */
    private String path;

    /**
     * 菜单组件
     */
    private String component;

    /**
     * 图标
     */
    private String icon;

    /**
     * 排序号
     */
    private Integer sort;

    /**
     * 是否可见
     */
    private Boolean visible;

    /**
     * 状态 0-禁用 1-启用
     */
    private Integer status;

    /**
     * API路径（用于API权限）
     */
    private String apiPath;

    /**
     * 请求方法（GET、POST、PUT、DELETE等）
     */
    private String method;

    /**
     * 备注
     */
    private String remark;
}
